src/containerapp/azext_containerapp/containerapp_env_telemetry_decorator.py

```diff
@@ -0,0 +1,72 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from azure.cli.command_modules.containerapp._utils import get_default_workload_profiles, safe_set
+from knack.log import get_logger
+
+from azure.cli.command_modules.containerapp.containerapp_env_decorator import ContainerAppEnvUpdateDecorator
+from azure.cli.core.azclierror import RequiredArgumentMissingError, ValidationError
+from ._utils import safe_get
+from ._client_factory import handle_non_404_status_code_exception
+
+from ._models import AppInsightsConfiguration, OpenTelemetryConfiguration, DestinationsConfiguration, \
+    TracesConfiguration, LogsConfiguration, MetricsConfiguration, DataDogConfiguration
+
+logger = get_logger(__name__)
+
+class ContainerappEnvTelemetryPreviewSetDecorator(ContainerAppEnvUpdateDecorator):
```

Comment:
I think this command should not extend `ContainerAppEnvUpdateDecorator`, because `ContainerAppEnvUpdateDecorator` is for command `az containerapp env update`, and arguments for command `env telemetry set` are not used in `ContainerAppEnvUpdateDecorator`.

```suggestion
class ContainerappEnvTelemetrySetDecorator(BaseResource):

```